data/transactions/logic/eval.go

```diff
@@ -216,6 +216,9 @@ type EvalParams struct {
 
 	// determines eval mode: runModeSignature or runModeApplication
 	runModeFlags runMode
+
+	// Total pool of app call budget in a group transaction
+	PooledApplicationBudget *uint64
```

Comment:
Please add a test near `prepareEvalParams` that checks budget value for 2 app calls in a group. That would ensure if someone changes it to a value a red flag will be raised as soon as possible.